pydantic/main.py

```diff
@@ -251,7 +251,11 @@ def __new__(mcs, name, bases, namespace):
             '_custom_root_type': _custom_root_type,
             **{n: v for n, v in namespace.items() if n not in fields},
         }
-        return super().__new__(mcs, name, bases, new_namespace)
+
+        # Necessary to not break smart debuggers when inspecting `self` like: WebPDB, VSCode, PyCharm..
+        obj = super().__new__(mcs, name, bases, new_namespace)
+        obj.__values__ = {}
```

Comment:
looks good, please can you add a test for this.

Something like

```py
def test_init_inspection():
    class Foobar(BaseModel):
        x: int

        def __init__(self, **data) -> None:
            with pytest.raises(AttributeError):
                assert self.x
            super().__init__(**data)

    Foobar(x=1)
```

Reply:
This worked out of the box